src/main/scala/akka/persistence/cassandra/query/journal/DeliveryBuffer.scala

```diff
@@ -0,0 +1,23 @@
+package akka.persistence.cassandra.query.journal
+
+import akka.stream.actor.ActorPublisher
+
+private[journal] trait DeliveryBuffer[T] { _: ActorPublisher[T] ⇒
+
+  def deliverBuf(buf: Vector[T]): Vector[T] =
```

Comment:
Yes, this immutable buffer is nice. One drawback is that I that think we should change to mutable `java.util.LinkedList` for optimal performance.